### ISSUE TITLE
Fix order of the gates from `draw(::Circuit)`

### ISSUE THREAD

**Issue by jofrevalles** [bsc-quantic/Quac.jl#15]
### Summary
The ordering of the gates in a `Circuit` displayed by the `draw(::Circuit)` function sometimes is not accurate.

### Example
```julia
julia> using Quac
julia> circ = Circuit(4); push!(circ, CX(1,3)); push!(circ, H(1)); push!(circ, H(2))
julia> circ
```
And the plot:
![image](https://user-images.githubusercontent.com/61060572/209137103-5bb228e2-1389-4e7f-8309-31e5f0767c77.png)

The priorities and moments seems to be fine:
```julia
julia> circ.lanes
4-element Vector{Vector{Quac.Element{Gate}}}:
 [Quac.Element{Gate}(Control{X}(1, X(3)), [1 => 1, 3 => 1]), Quac.Element{Gate}(H(1), [1 => 2])]
 [Quac.Element{Gate}(H(2), [2 => 1])]
 [Quac.Element{Gate}(Control{X}(1, X(3)), [1 => 1, 3 => 1])]
 []

julia> moments(circ)
2-element Vector{Vector{Gate}}:
 [Control{X}(1, X(3)), H(2)]
 [H(1)]
```

**Comment by mofeing** [comment]
Ah yes. This the fault of the `draw` function.

The problem is that although `CX(1,3)` and `H(2)` "live" in the same moment (this is correct behavior), they cannot be drawn in the same "moment" because of overlapping drawings. One of the two needs to be moved to the next moment.

Current solution is to create a "ghost" moment where the overlapping gate is moved (i.e. `H(2)` in the drawing). A better solution would be to do some kind of circuit compilation.

**Comment by jofrevalles** [comment]
I think this behavior of the `H(2)` is the expected for visualization, since it is true that it overlaps with `CX(1, 3)`. Nevertheless, I would think that this shift of `H(2)` should not affect the rest of the gates (e.g., `H(1)` in this example).

**Comment by mofeing** [comment]
I agree with you but then you need to compare between moments and do some kind of basic compilation.

**Comment by jofrevalles** [comment]
Also sometimes I find that the `draw` function draws empty gates when these are not necessary. As an example:

```julia
julia> circ = Circuit(4); push!(circ, Sd(1)); push!(circ, CX(2,4)); push!(circ, CZ(1,4)); push!(circ, X(1))
julia> circ
```
Produces:
![image](https://user-images.githubusercontent.com/61060572/209782813-17869505-55b8-43a8-b401-f1af61183c89.png)

**Comment by mofeing** [comment]
mmm strange. this is a bug also.
